src/core/symbology-ng/qgssymbollayerutils.cpp

```diff
@@ -3986,8 +4003,26 @@ double QgsSymbolLayerUtils::rescaleUom( double size, QgsUnitTypes::RenderUnit un
           scale = 1 / 0.28;
           roundToUnit = true;
           break;
-        // we don't have a good case for map units, as pixel values won't change based on zoom
-        default:
+        case QgsUnitTypes::RenderInches:
+          scale = 1 / 0.28 * 25.4;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPoints:
+          scale = 90. /* dots per inch according to OGC SLD */ / 72. /* points per inch */;
+          roundToUnit = true;
+          break;
+        case QgsUnitTypes::RenderPixels:
+          // pixel is pixel
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderMapUnits:
+        case QgsUnitTypes::RenderMetersInMapUnits:
+          // already handed via uom
+          scale = 1;
+          break;
+        case QgsUnitTypes::RenderPercentage:
+        case QgsUnitTypes::RenderUnknownUnit:
+          // these do not make sense and should not really reach here
```

Comment:
@luipir you comment is stated like an explanation, but reading it a few times, it also looks like a request for change... which one is it?

The SLD export code has several places in which it does not know or cannot do anything sane and it's already being tolerant instead of throwing exceptions.
If an exception is thrown here, what is going to happen to QGIS? Hard crash, nice user error report? And how does one control the context of the exception, just saying "there is a unit in your style I cannot handle" without saying where it is, that's quite mean.